nrfx/hal/nrf_ficr.h

```diff
@@ -151,6 +193,42 @@ NRF_STATIC_INLINE uint32_t nrf_ficr_nfc_tagheader_get(NRF_FICR_Type const * p_re
                                                       uint32_t              tagheader_id);
 #endif
 
+#if NRF_FICR_HAS_DEVICE_ADDR || NRF_FICR_HAS_BLE_ADDR
+/**
+ * @brief Function for getting the unique device address.
+ *
+ * @param[in] p_reg  Pointer to the structure of registers of the peripheral.
+ * @param[in] reg_id Register index.
+ *
+ * @return Unique device address.
+ */
+NRF_STATIC_INLINE uint32_t nrf_ficr_deviceaddr_get(NRF_FICR_Type const * p_reg, uint32_t reg_id);
```

Comment:
is it a good idea to abstract at this HAL level? should we just have `nrf_ficr_deviceaddr_get` and `nrf_ficr_bleaddr_get`, and use one or the other at driver level based on platform?

Reply:
This is done for consistency with existing helpers, like the one for device id.